src/common/snippets/include/snippets/lowered/loop_manager.hpp

```diff
@@ -266,6 +266,14 @@ class LoopManager {
      */
     LoopPort get_loop_port_by_expr_port(const ExpressionPort& expr_port, const size_t loop_id);
 
+    /**
+     * @brief Reassign Loop IDs in `m_map` using `loop_id_map` and update LoopIDs of expressions and LoopEnd
+     * @param linear_ir linear IR
+     * @param loop_id_map [ current Loop ID -> new target Loop ID ]
+     * @return True if loops have been reassigned. Otherwise returns False
+     */
+    bool reassign_identifiers(const LinearIR& linear_ir, const std::map<size_t, size_t>& loop_id_map);
```

Comment:
`reassign_identifiers` is used only in `NormalizeLoopIDs`, and there we need only reorder some existing loops.
And we have a very similar methods in `UnifiedLoopInfo` that do exactly that: `sort_input_ports` and `sort_output_ports`.
They are safer (since don't allow arbitrary changing the ids) and simpler. Also, it's kind of makes sense to align interface on different levels: loop_info can reorder ports, and loop_manager can reorder loops.

What do you think, do we really need such a generic interface for loop reassignment, or simple reordering would be enough?

I do not insist that we do it in this PR, to be clear.

Reply:
I tried to implement generic algorithm for `LoopManager`. However seems like to implement `reorder_identifiers` is enough for now. Thus, we will align with `reorder_loop_ports` of `LoopInfo`.
I replaced `reassign_identifiers` with `reorder_indetifiers`, thanks